### DISCUSSION TITLE
issues with /v13/deployments

### DISCUSSION THREAD

**Discussion by cjoejoe1** [vercel/vercel#8440] [Help]
Hi,

I'm using the Vercel API to create a project and then deploy the project.

A) I use /v9/projects to create a new project and that works fine.
B) However when I create a new command to deploy/build the project I get an error.

A) Create the new project with a name of newproject9 > Works great.

          const createProject = await axios.post(`https://api.vercel.com/v9/projects/`,{
          'name': 'newproject9',
          'buildCommand': 'npm run build',
          'installCommand': 'npm install',
          'framework': 'nextjs',
          'gitRepository': {
            "repo": 'XXXXXX',
            "type": "github"
          }
          }, getRequestConfig())

B) Deploy the project with the name "newproject9" >> This gives an error message :  {"severity":"WARNING","message":"Function returned undefined, expected Promise or value"}

`       const deploy = await axios.post('https://api.vercel.com/v13/deployments?forceNew=1',{
              "name": "newproject9",
            }, getRequestConfig())`

I feel like i've missed out something quite obvious, any help would be great!

Thanks,

**Comment by amyegan** [comment]
Thanks for reaching out @cjoejoe1! It should work if you include `gitSource` or `files` to tell the deployment what files it should use.

**Reply by cjoejoe1** [reply]
Hi @amyegan, thanks for getting back to me so quick!

Is this what you mean? I'm still getting the same error:

 `const deploy = await axios.post('https://api.vercel.com/v13/deployments?forceNew=1',{
              "name": "newproject9",
              "gitSource": {
                "type": "github",
                "repoId": "XXXXXXX"
              }
            }, getRequestConfig())`

  {"severity":"WARNING","message":"Function returned undefined, expected Promise or value"} 
 Request failed with status code 400

The same format to create the project works fine.

This is probably a stupid question but shouldn't 

`https://api.vercel.com/v13/deployments`

return
"The request is missing an authentication token" 

and not: 
'invalid API version" 

It feels like the v13 is not active ?

Thanks again !

**Reply by amyegan** [reply]
Not a stupid question at all! You'll get the "invalid API version" error if you try to use an endpoint that isn't yet at that version. So a POST to create a new deployment at `/v13/deployments` should work, but `/v13/projects` will throw the error. If you want to [GET a list of all deployments](/v6/deployments), you should use `/v6/deployments`.

I just tried your API calls and it worked for me. The only thing I changed from your example was my test repo id and a `ref` to the last buildable commit in my test repo in `gitSource`. Do you get more details if you run it with Postman or `curl`? That might give us more clues about what's happening.
